bitwarden_license/bit-common/src/dirt/reports/risk-insights/services/risk-insights-report.service.ts

```diff
@@ -162,6 +174,106 @@ export class RiskInsightsReportService {
     };
   }
 
+  async generateEncryptedRiskInsightsReport(
+    organizationId: OrganizationId,
+    details: ApplicationHealthReportDetail[],
+    summary: ApplicationHealthReportSummary,
+  ): Promise<RiskInsightsReport> {
+    const orgKey = await this.keyService.getOrgKey(organizationId as string);
+    if (orgKey === null) {
+      throw new Error("Organization key not found");
+    }
+
+    const reportContentEncryptionKey = await this.keyGeneratorService.createKey(512);
+    const reportEncrypted = await this.encryptService.encryptString(
+      JSON.stringify(details),
+      reportContentEncryptionKey,
+    );
+
+    const wrappedReportContentEncryptionKey = await this.encryptService.wrapSymmetricKey(
+      reportContentEncryptionKey,
+      orgKey,
+    );
+
+    const reportDataWithWrappedKey = {
+      data: reportEncrypted.encryptedString,
+      key: wrappedReportContentEncryptionKey.encryptedString,
+    };
+
+    const encryptedReportDataWithWrappedKey = await this.encryptService.encryptString(
+      JSON.stringify(reportDataWithWrappedKey),
+      orgKey,
+    );
+
+    const criticalApps = await firstValueFrom(
+      this.criticalAppsService
+        .getAppsListForOrg(organizationId)
+        .pipe(takeWhile((apps) => apps !== null && apps.length > 0)),
+    );
+
+    const riskInsightReport = {
+      organizationId: organizationId,
+      date: new Date().toISOString(),
+      reportData: encryptedReportDataWithWrappedKey.encryptedString,
+      totalMembers: summary.totalMemberCount,
+      totalAtRiskMembers: summary.totalAtRiskMemberCount,
+      totalApplications: summary.totalApplicationCount,
+      totalAtRiskApplications: summary.totalAtRiskApplicationCount,
+      totalCriticalApplications: criticalApps.length,
+    };
+
+    return riskInsightReport;
+  }
+
+  async decryptRiskInsightsReport(
+    organizationId: OrganizationId,
+    riskInsightsReportResponse: GetRiskInsightsReportResponse,
+  ): Promise<[ApplicationHealthReportDetail[], ApplicationHealthReportSummary]> {
+    try {
+      const orgKey = await this.keyService.getOrgKey(organizationId as string);
+      if (orgKey === null) {
+        throw new Error("Organization key not found");
+      }
+
+      const decryptedReportDataWithWrappedKey = await this.encryptService.decryptString(
+        new EncString(riskInsightsReportResponse.reportData),
+        orgKey,
+      );
+
+      const reportDataInJson = JSON.parse(decryptedReportDataWithWrappedKey);
+      const reportEncrypted = reportDataInJson.data;
+      const wrappedReportContentEncryptionKey = reportDataInJson.key;
```

Comment:
```suggestion
const wrappedReportContentEncryptionKey = riskInsightsReportResponse.reportKey;
const reportEncrypted = riskInsightsReportResponse.reportData;
```

With the same reasoning as above, I don't think we need two layers of encryption here?